projects/igniteui-angular/src/lib/services/overlay/overlay.ts

```diff
@@ -330,9 +331,7 @@ export class IgxOverlayService implements OnDestroy {
             wrapperElement.classList.add('igx-overlay__wrapper--modal');
         }
 
-        if (info.settings.closeOnEsc) {
-            this.setUpCloseOnEscape(info);
-        }
+        this.setUpCloseOnEscape();
```

Comment:
rename this to `addCloseOnEscapeListener` just to be consistent with the other two methods and move the call just after `addResizeHandler`
Let's make our code better :)